fastparquet/api.py

```diff
@@ -416,19 +416,23 @@ def to_pandas(self, columns=None, categories=None, filters=[],
             inferred from the metadata (if this was originally pandas data); if
             the metadata does not exist or index is False, index is simple
             sequential integers.
-        row_filter: bool
+        row_filter: bool or tuple of a list and a ndarray
             Whether filters are applied to whole row-groups (False, default)
             or row-wise (True, experimental). The latter requires two passes of
             any row group that may contain valid rows, but can be much more
             memory-efficient, especially if the filter columns are not required
             in the output.
+            If a tuple, the first element has to be the list of row groups to
```

Comment:
List of the row-group objects or list of indices? I wonder if we need this, since you can make new ParquetFile instances by selecting row-groups already.

Reply:
I opted for list of row group objects as it seems to me the code in `to_pandas()` is then one line shorter.

If we accept list of indices, as I picture it, we have then to restore the subset of row groups by list comprehension in `to_pandas()`.
Eg
```python
rgs = [self.row_groups[i] for i in row_filter[1]]
```

On my side, I extracted from previous implementation (in which `n_prev_rows` was used) the code to generate this row group subset + row per row boolean array. I have put this in a function that outputs this data.
Outputting list of row group objects or list of indices is the same (only a parameter to change in `filter_row_groups()`).

But as said above, in `to_pandas()`, accepting list of indices requires then one line more of code to get back the list of row group objects.

I would keep current code because of this.
If you feel we should stick to list of indices, I can make the change, but at the moment, I don't see the advantage in it?

Reply:
> If you feel we should stick to list of indices

I was looking for clarification, I have no preference

Reply:
> I wonder if we need this, since you can make new ParquetFile instances by selecting row-groups already.

I actually did not understand this during our previous discussion, I am sorry.
Thanks for raising the point.
Your thought may be simple, but reading advices needs (at least for me) double or triple readings :)

I fully agree with you. I tested this version in my targeted use case (adding `n_prev_rows`) and this does not impact impact negatively the code, on the opposite, I could suppress one row! :)

So my last commit also take this comment into account.